File: examnation/lib/models/past_exam.dart
class PastExam {
  final int id;
  final String title;
  final String subject;
  final String level;
  final int year;
  final String filePath;
  final String? fileType;

  PastExam({
    required this.id,
    required this.title,
    required this.subject,
    required this.level,
    required this.year,
    required this.filePath,
    this.fileType,
  });

  factory PastExam.fromJson(Map<String, dynamic> json) => PastExam(
    id: json['id'] is int ? json['id'] : int.parse(json['id'].toString()),
    title: json['title'] ?? '',
    subject: json['subject'] ?? '',
    level: json['level'] ?? '',
    year: json['year'] is int
        ? json['year']
        : int.parse(json['year'].toString()),
    filePath: json['filePath'] ?? json['file_path'] ?? '',
    fileType: json['fileType'] ?? json['file_type'],
  );
}

File: backend/src/main/java/com/examnation/backend/model/PastExam.java
package com.examnation.backend.model;

import lombok.Data;
import jakarta.persistence.*;
import java.time.Year;

@Entity
@Table(name = "past_exams")
@Data
public class PastExam {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column(nullable = false)
    private String subject;

    @Column(nullable = false)
    private Year year;

    @Column(nullable = false)
    private String level;

    @Column(nullable = false)
    private String filePath;

    @Column(nullable = false)
    private String fileType;
}